src/dmsort.rs

```diff
@@ -217,13 +217,14 @@ impl<'a, T> Drop for DmSorter<'a, T> {
 unsafe fn unsafe_push<T>(vec: &mut Vec<T>, value: &T) {
 	let old_len = vec.len();
 	vec.reserve(1);
-	ptr::copy_nonoverlapping(value, vec.get_unchecked_mut(old_len), 1);
+	ptr::copy_nonoverlapping(value, vec.as_mut_ptr().add(old_len), 1);
```

Comment:
This `get_unchecked_mut` is always out-of-bounds by definition. If you can tolerate a recent Rust version, this could also be implemented with `&mut vec.spare_capacity_mut()[0]`, though I think pointer offsetting alone is totally fine here.